### ISSUE TITLE
Excluding SSIDs / only using DNS for cellular not 100% working

### ISSUE THREAD

**Issue by GER-HerrS** [fyr77/dns-mobileconfig#17]
First of all: Thanks for this great tool!
Please excuse my english as it is not always the best.
Just ask if something is not clear.

Unfortunately I have some problems with my config and I don't know if there is a problem with the config itself or did I just generate it wrong. For this reason I put my problem here.

I am using Pi-Hole in two Wi-Fis so I want my iPhone to use the Pi-Hole IP as DNS when it is connected to one of those SSIDs (over DHCP).
Lets call them MyWLAN1 and MyWLAN2.
In case my iPhone is connected to another Wi-Fi or is just surfing through cellular I want the iPhone to use the "Digitale Gesellschaft DNS".
I got the informations concerning "Digitale Gesellschaft" from here:
https://github.com/DigitaleGesellschaft/DNS-Resolver#technical-information--configuration-how-tos 

So I used the tool and downloaded my profile using all IPv4 and IPv6 adresses and the DoT URL as well.
Also I declared the two excluded Wi-Fis.
Afterwards I sent the profile to my iPhone via e-mail, installed and used it.

Now all worked fine when I was connected to my MyWLAN1.
Turning the Wi-Fi off my internet didn't work any more. Deleting the profile helped me out and I could surf the internet.
So in conclusion I can say the Wi-Fi part did work. But the cellular part didn't - and I have no idea why.

Then I tried to use a profile that only uses encrypted DNS with cellular interface (no Wi-Fi, no excludes).
It was the same problem.

Here you can get my custom profiles. I changed the original SSIDs via Notepad++ to "MyWLAN1" and "MyWLAN2".
In fact the SSIDs have other names that I used in the tool when creating the profile.

[dns-profile-issue.zip](https://github.com/fyr77/dns-mobileconfig/files/7519511/dns-profile-issue.zip)

Thanks for looking over it.

Greetings from Germany
HerrS

UPDATE:
Maybe it is useful to say that I use an iPhone 12 with actual iOS 15.1.

**Comment by fyr77** [comment]
Hello!
I'm happy my tool is of use to you! I took a look at your config files and noticed, that you added the port (:853) to the DoT server name. Could you try again without the port specified? It is the default port anyway.

Here are your files with the port removed: 
[dns-cellular-test.zip](https://github.com/fyr77/dns-mobileconfig/files/7520328/dns-cellular-test.zip)

Please tell me if that works for you.

**Comment by GER-HerrS** [comment]
Hi!
Indeed this helped (of course I had to change the Wi-Fi SSIDs to the correct ones - first time I tested your config I forgot).
So I suggest you modify the tool to delete the ":853" if a user writes it into the url field (at least for this standard port ; no idea if another port will actually work).

Many thanks for your quick and successful help.

**Comment by fyr77** [comment]
Wonderful, I'll implement a check for that.
